hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -94,6 +101,18 @@ public BaseSparkCommitActionExecutor(HoodieEngineContext context,
                                        WriteOperationType operationType,
                                        Option extraMetadata) {
     super(context, config, table, instantTime, operationType, extraMetadata);
+    initKeyGenIfNeeded();
+  }
+
+  private void initKeyGenIfNeeded() {
+    this.populateMetaFields = config.populateMetaFields();
+    if (!populateMetaFields) {
+      try {
+        keyGeneratorOpt = Option.of((BaseKeyGenerator) HoodieSparkKeyGeneratorFactory.createKeyGenerator(new TypedProperties(config.getProps())));
+      } catch (IOException e) {
+        throw new HoodieIOException("Only BaseKeyGenerators are supported when meta columns are disabled ", e);
```

Comment:
move this exception handling  into the method itself? its an unchecked exception anyway. we can save some lines

Reply:
not very sure on this. Did you mean to suggest to move this to the caller of this method or within HoodieSparkKeyGeneratorFactory.createKeyGenerator. 
Bcoz, we have two constructors and to avoid duplicate code, I am doing this in private method. 
Also, we can't move it within HoodieSparkKeyGeneratorFactory.createKeyGenerator, bcoz, here we are casting it to BaseKeyGenerator since, if meta fields are disabled, we have some constraints around keygens.

Reply:
>within HoodieSparkKeyGeneratorFactory.createKeyGenerator.

yes. within. You can still cast outside right?

Reply:
But I added as a guard for catching casting issues.